controller_interface/src/controller_interface_base.cpp

```diff
@@ -71,6 +73,7 @@ return_type ControllerInterfaceBase::init(
   node_->register_on_activate(
     [this](const rclcpp_lifecycle::State & previous_state) -> CallbackReturn
     {
+      enable_introspection(true);
```

Comment:
I thought there would be a new parameter users could set in their `control.yaml` or `xacro` to choose if they wanted this enabled. Do you see value in letting users choose if they want to enable/disable this feature?

Reply:
The idea is to have the introspection of the variables when the controller and hardware is active. As soon as it is inactive it will be deactivated by default.
The users can still call the `enable_introspection(true);` in their configure or on_deactivate methods to increase the lifetime of the introspection.

Moreover, the topic information is not published unless there is a subscriber.